src/app/courses/pages/new-course-page/new-course-page.component.ts

```diff
@@ -10,19 +13,36 @@ import { Router } from '@angular/router';
 export class NewCoursPageComponent implements OnInit {
 
   formNewCourse!: FormGroup;
+  course: Course;
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private coursesService: CoursesService,
+     ) {}
 
-  constructor(private router: Router) { }
+  title: string = 'New course';
 
   ngOnInit(): void {
     this.buildForm();
+    this.route.params.subscribe(
+      (params: Params) => {
+        if (params['id']){
+          this.title = 'Edit course'
+          this.course = this.coursesService.getCourse(+params['id']);
+        }
+      }
+    );
   }
 
+
   onSubmit(): void {
 
   }
 
   createCourse(): void {
     this.router.navigate(['/courses']);
+    this.coursesService.updateCourse(this.course);
```

Comment:
It should call `createCourse()` when you add a new course and call `updateCourse(this.course)` when you update the existing course.